crates/rolldown_binding/src/options/plugin/js_plugin.rs

```diff
@@ -32,10 +35,18 @@ pub struct JsPluginContextResolveCustomArgId;
 impl TypedMapKey for JsPluginContextResolveCustomArgId {
   type Value = u32;
 }
+#[derive(Debug, PartialEq, Eq)]
+pub enum FilterExprCacheKey {
+  ResolveId,
+  Transform,
+  Load,
+}
 
 #[derive(Debug)]
 pub struct JsPlugin {
   pub(crate) inner: BindingPluginOptions,
+  /// Since there at most three key in the cache, use vec should always faster than hashmap
+  pub(crate) filter_expr_cache: Vec<(FilterExprCacheKey, Vec<FilterExprKind>)>,
```

Comment:
Not from performance aspect. Would it be more straight if you just put them like

``` rust
struct JsPlugin {
  transform_filter: ...,
  load_filter: ...,
  resolve_id_filter: ...,
}
```

since you are already do the enumerating in `FilterExprCacheKey`.